app/src/main/java/com/islaharper/dawnofandroid/presentation/common/OneTap.kt

```diff
@@ -0,0 +1,133 @@
+package com.islaharper.dawnofandroid.presentation.common
+
+import android.app.Activity
+import android.util.Log
+import androidx.activity.compose.ManagedActivityResultLauncher
+import androidx.activity.compose.rememberLauncherForActivityResult
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.IntentSenderRequest
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.platform.LocalContext
+import com.google.android.gms.auth.api.identity.BeginSignInRequest
+import com.google.android.gms.auth.api.identity.Identity
+import com.google.android.gms.common.api.ApiException
+import com.google.android.gms.common.api.CommonStatusCodes
+import com.islaharper.dawnofandroid.util.Constants.CLIENT_ID
+
+@Composable
+fun StartActivityForResult(
```

Comment:
Is this code based on a sample somewhere (e.g., Stack Overflow or in a Udemy course)? If so, can you please link it so the reviewer has more context to understand the design of the solution?

Reply:
I reviewed a number of different methods for One Tap sign in, this looked like a good option for use in Compose. I will try find the original article to reference.